cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +795,83 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_mask_first_n(int64_t context, const char* data, int32_t data_len,
+                                int32_t n_to_mask, int32_t* out_len) {
+  if (data_len <= 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  std::string str(data, data_len);
+  int32_t counter = 0;
+  for (char& c : str) {
+    if (n_to_mask > 0 && counter < n_to_mask && isalnum(c)) {
+      if (isdigit(c)) {
+        c = 'n';
+      } else {
+        if (isupper(c)) {
+          c = 'X';
+        } else {
+          c = 'x';
+        }
+      }
+      counter++;
+    }
+  }
+
+  *out_len = static_cast<int32_t>(str.size());
+
+  char* out = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, *out_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+
+  memcpy(out, str.c_str(), *out_len);
+  return out;
+}
+
+GANDIVA_EXPORT
+const char* gdv_fn_mask_last_n(int64_t context, const char* data, int32_t data_len,
```

Comment:
Please change this to be similar to mask_first_n

Use memcpy for the last_n; and the array for the digits that need masking

Reply:
Did the same improvements on mask_last_n